File: Barone.api/Models/BarrilModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Web;

namespace Barone.api.Models
{
    public class BarrilModel
    {
        [Key]
        public int id { get; set; }
        public string NroBarril { set; get; }
        public int idEstado { set; get; }
        public int? IdEstilo { get; set; }
        public string CantidadLitros { set; get; }
        public long? idEntrega { set; get; }
        public EstilosModel Estilo { get; set; }
        public MovimientosModel Entrega { set; get; }
        [ForeignKey("Coccion_id")]
        public virtual CoccionModel Coccion { get; set; }
        public int? Coccion_id { set; get; }




    }
}

File: Barone.Client/src/app/modules/shared/models/barril.model.ts
import { Constructable } from "../../../core/models/constructable";
import { EstilosModel } from "./estilos.model";
import { CoccionModel } from "./coccion/coccion.model";
import { EntregaModel } from "./entrega.model";

export class BarrilModel extends Constructable<Partial<BarrilModel>>{
    id: number;
    NroBarril: string;
    idEstado: number;
    CantidadLitros: string;
    idEntrega?: number;
    IdEstilo?: number;
    Estilo?: EstilosModel;
    Coccion_id?: number;
    Coccion?: CoccionModel;
    Entrega? :EntregaModel;
}